ios/web_view/public/cwv_web_view_extras.h

```diff
@@ -0,0 +1,77 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_IOS_WEB_VIEW_PUBLIC_CWV_WEB_VIEW_EXTRAS_H_
+#define BRAVE_IOS_WEB_VIEW_PUBLIC_CWV_WEB_VIEW_EXTRAS_H_
+
+#import <Foundation/Foundation.h>
+#import <UIKit/UIKit.h>
+#import <WebKit/WebKit.h>
+
+#import "cwv_export.h"
+#import "cwv_web_view.h"
+
+typedef NSInteger CWVUserAgentType NS_TYPED_ENUM;
+OBJC_EXPORT const CWVUserAgentType CWVUserAgentTypeNone;
+OBJC_EXPORT const CWVUserAgentType CWVUserAgentTypeAutomatic;
+OBJC_EXPORT const CWVUserAgentType CWVUserAgentTypeMobile;
+OBJC_EXPORT const CWVUserAgentType CWVUserAgentTypeDesktop;
+
+NS_ASSUME_NONNULL_BEGIN
+
+CWV_EXPORT
+@interface CWVWebView (Extras)
```

Comment:
why is this here, this seems to be a copy of the upstream file?

Reply:
These are not copies of any upstream files, they provide additional support that is available via WebState but not available via CWVWebView

Reply:
sorry, actually this seems to have the methods from CRWWebController so it's not clear why they're on the webview here.

Reply:
whatever the reason, this kind of thing should definitely have comments explaining